x-pack/solutions/security/plugins/security_solution/public/siem_migrations/rules/components/migration_status_panels/migration_progress_panel.test.tsx

```diff
@@ -0,0 +1,115 @@
+ * 2.0.
+ */
+
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { MigrationProgressPanel } from './migration_progress_panel';
+import { useStopMigration } from '../../service/hooks/use_stop_migration';
+import { SiemMigrationTaskStatus } from '../../../../../common/siem_migrations/constants';
+import { TestProviders } from '../../../../common/mock';
+import type { RuleMigrationStats } from '../../types';
+
+jest.mock('../../../../common/lib/kibana/use_kibana');
+
+jest.mock('../../service/hooks/use_stop_migration');
+const useStopMigrationMock = useStopMigration as jest.Mock;
+const mockStopMigration = jest.fn();
+
+const inProgressMigrationStats: RuleMigrationStats = {
+  status: SiemMigrationTaskStatus.RUNNING,
+  id: 'c44d2c7d-0de1-4231-8b82-0dcfd67a9fe3',
+  rules: { total: 26, pending: 6, processing: 10, completed: 9, failed: 1 },
+  created_at: '2025-05-27T12:12:17.563Z',
+  last_updated_at: '2025-05-27T12:12:17.563Z',
+  number: 1,
+};
+const preparingMigrationStats: RuleMigrationStats = {
+  ...inProgressMigrationStats,
+  // status RUNNING and the same number of total and pending rules, means the migration is still preparing the environment
+  status: SiemMigrationTaskStatus.RUNNING,
+  rules: { total: 6, pending: 6, processing: 0, completed: 0, failed: 0 },
+};
+
+const renderMigrationProgressPanel = (migrationStats: RuleMigrationStats) => {
+  return render(<MigrationProgressPanel migrationStats={migrationStats} />, {
+    wrapper: TestProviders,
+  });
+};
+
+describe('MigrationProgressPanel', () => {
+  beforeEach(() => {
+    useStopMigrationMock.mockReturnValue({
+      stopMigration: mockStopMigration,
+      isLoading: false,
+    });
+  });
+
+  describe('Preparing migration', () => {
+    beforeEach(() => {
+      renderMigrationProgressPanel(preparingMigrationStats);
+    });
+
+    it('should render description text correctly', () => {
+      expect(screen.queryByTestId('ruleMigrationDescription')).toHaveTextContent(
+        `Preparing environment for the AI powered translation.`
+      );
+    });
+
+    it('should render spinner', () => {
+      expect(screen.queryByTestId('ruleMigrationSpinner')).toBeInTheDocument();
+    });
+
+    it('should not render progress bar', () => {
+      expect(screen.queryByTestId('ruleMigrationProgressBar')).not.toBeInTheDocument();
+    });
+  });
+
+  describe('In progress Migration', () => {
+    beforeEach(() => {
+      renderMigrationProgressPanel(inProgressMigrationStats);
+    });
+
+    it('should render description text correctly', () => {
+      expect(screen.getByTestId('ruleMigrationDescription')).toHaveTextContent(`Translating rules`);
+    });
+
+    it('should render spinner', () => {
+      expect(screen.queryByTestId('ruleMigrationSpinner')).toBeInTheDocument();
+    });
+
+    it('should render progress bar', () => {
+      expect(screen.queryByTestId('ruleMigrationProgressBar')).toBeInTheDocument();
```

Comment:
same here. Isn't it better to be more precise with `toBeVisible`  than `toBeInTheDocument`?
```suggestion
      expect(screen.queryByTestId('ruleMigrationProgressBar')).toBeVisible();
```

Reply:
We don't usually render hidden components. I think it's not a very good practice and should be avoided if possible. We only do that in the Timeline so users have it always ready and loaded, but it has also been the source of many issues.
Let's stick to `toBeInTheDocument`.